### PR TITLE
Superformula - Venkatesh

### PR DESCRIPTION
**Pull request by venkatesh-seelam**
Test Code has been completed

### CODE REVIEW

File: lib/providers/restaurantsprovider.dart [hunk 1]
```diff
@@ -0,0 +1,90 @@
+/// Copyright 2022 - Superformula. All rights reserved.
+import 'package:flutter/material.dart';
+import 'package:restaurantour/repositories/yelp_repository.dart';
+import 'package:restaurantour/models/restaurant.dart';
+
+/// This is a Provider clss which extends ChangeNotifies to mange the Sate for the Restaurants.
+class RestaurantsProvider extends ChangeNotifier {
```
Comment (jamesshaw-sf) [inline lib/providers/restaurantsprovider.dart:7]: suggestion: file should have an underscore between words.

File: lib/providers/restaurantsprovider.dart [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  Map get allRestaurantsMap => _allRestaurants;
+  Map get allFavouriteRestaurants => _restMap;
+  bool get isFirstLoading => _firstLoading;
+  int get allRestaurantsListCount => _allRestaurantsList.length;
+
+  /// Load all Restaurants from the given Yelp API
+  /// If the given listener is not registered, the call is ignored.
+  /// This method must not be called after [dispose] has been called.
+  ///
+
```
Comment (jamesshaw-sf) [inline lib/providers/restaurantsprovider.dart:23]: suggestion: there shouldn't be a new line here.

File: lib/widgets/restaurantlistitem.dart [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+            decoration: BoxDecoration(
+              borderRadius: const BorderRadius.all(Radius.circular(8)),
+              image: DecorationImage(image: NetworkImage(image), fit: BoxFit.fill),
+            ),
+          ),
+          Expanded(
+              child: Container(
+                  height: 88,
+                  margin: const EdgeInsets.only(left: 5.0, right: 15.0, top: 10.0, bottom: 10.0),
+                  child: Column(crossAxisAlignment: CrossAxisAlignment.start, children: [
```
Comment (jamesshaw-sf) [inline lib/widgets/restaurantlistitem.dart:50]: bug: overflow.
Screen Shot 2022-07-19 at 9 10 44 AM

File: lib/widgets/restaurantdetails.dart [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+                        height: 50, //color: Colors.lightGreen,
+                        child: CircularProgressIndicator(valueColor: AlwaysStoppedAnimation<Color>(Colors.black))),
+                  )
+                : Scaffold(
+                    appBar: AppBar(
+                      elevation: 0,
+                      title: Text(restDetails.name),
+                      actions: <Widget>[
+                        Padding(
+                            padding: EdgeInsets.only(right: 20.0),
```
Comment (jamesshaw-sf) [inline lib/widgets/restaurantdetails.dart]: chore: could be const.

File: lib/widgets/allrestaurants.dart [hunk 5]
```diff
@@ -0,0 +1,69 @@
+/// Copyright 2022 - Superformula. All rights reserved.
```
Comment (jamesshaw-sf) [inline lib/widgets/allrestaurants.dart:1]: chore: The designs call for more padding on this screen.

File: lib/widgets/restaurantdetails.dart [hunk 6]
```diff
@@ -0,0 +1,157 @@
+/// Copyright 2022 - Superformula. All rights reserved.
+import "package:flutter/material.dart";
+import 'package:provider/provider.dart';
+import 'package:restaurantour/providers/restaurantsprovider.dart';
+
+/// A widget that displays Restaurant details in a SingleChildScrollView.
+/// A widget that does not require mutable state.
+/// Show all details of the selected Restaurant by calling the Provider
+/// This Class is route from the All Restaurants/Favourite Restaurants
+///
```
Comment (jamesshaw-sf) [inline lib/widgets/restaurantdetails.dart]: chore: remove the extra ///

File: lib/widgets/restaurantdetails.dart [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+                                      fontSize: 25,
+                                      color: Colors.black,
+                                    ),
+                                  ),
+                                  Container(
+                                      height: 25, alignment: Alignment.bottomLeft, child: const Icon(Icons.star, color: Colors.orange, size: 12))
+                                ])),
+                            const Divider(color: Colors.black),
+                            Container(alignment: Alignment.centerLeft, height: 50, child: Text("${restDetails.reviews.length} Reviews")),
+                            Container(
```
Comment (jamesshaw-sf) [inline lib/widgets/restaurantdetails.dart]: chore: this container isn't needed.

File: lib/widgets/restaurantdetails.dart [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+                                var review = restDetails.reviews[index];
+                                return Column(children: [
+                                  Row(
+                                      children: List.generate(review.rating.round(), (rindex) {
+                                    return const Icon(Icons.star, color: Colors.orange, size: 12);
+                                  })),
+                                  Container(
+                                    alignment: Alignment.centerLeft,
+                                    height: 45,
+                                    child: const Text("Review not posted"),
```
Comment (jamesshaw-sf) [inline lib/widgets/restaurantdetails.dart]: bug:  I don't see how the actual review text would ever be displayed.

File: lib/widgets/restaurantdetails.dart [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+                                          size: 10,
+                                          color: Colors.green,
+                                        )
+                                      : const Icon(Icons.circle_rounded, size: 10, color: Colors.red)
+                                ])
+                              ]),
+                            ),
+                            const Divider(color: Colors.black),
+                            Container(alignment: Alignment.centerLeft, height: 50, child: const Text("Address")),
+                            Container(alignment: Alignment.centerLeft, height: 50, child: Text(restDetails.location.formattedAddress ?? "")),
```
Comment (jamesshaw-sf) [inline lib/widgets/restaurantdetails.dart:96]: bug: the address is not displayed.
Screen Shot 2022-07-19 at 9 23 44 AM

File: lib/main.dart [hunk 10]
```diff
@@ -1,10 +1,98 @@
...
+import 'package:restaurantour/utils/properties.dart';
+import 'package:restaurantour/providers/restaurantsprovider.dart';
+import 'package:restaurantour/widgets/allrestaurants.dart';
+import 'package:restaurantour/widgets/favouriterestaurants.dart';
+import 'package:restaurantour/widgets/restaurantdetails.dart';
+
+///Start method - which is the main
+/// This is going to call the StatelessWidget
+/// runApp is the driver of the App
+void main() => runApp( RestaurantHome());
```
Comment (jamesshaw-sf) [inline lib/main.dart]: chore: remove extra space.

**Review by jamesshaw-sf — COMMENTED** [review]
Thanks for taking the time to submit your code for review!  I have left a few comments.

**Comment by venkatesh-seelam** [comment]
Hi James,

   I have fixed all issues except Reviews and Address.
   Address always returns null and Review class and User subclass doesn't
have any property for the review text.

  Code has been merged in Git and sent a pull request.

Thanks
Venkatesh

**Pull request closed without merging**